### DISCUSSION TITLE
Logs: Colorize log lines according to query

### DISCUSSION THREAD

**Discussion by Morriz** [grafana/grafana#56844] [Observability]
It is now not possible to "see" which query resulted in what log lines. In order to make sense of log lines that come from multiple sources those need to be identifiable visually. I believe a story for this was already started long time ago, and I was involved in that and we actually already agreed on implementing that. I can't find that issue here, but we agreed that grafana would allow us to choose a color per query, and the corresponding log lines coming from that query would be matched and colorized accordingly (css matching probably). An extra filter "Source" could be toggled (on by default for multi query) resulting in an extra column "Source" that shows the word in the right color.

**Steps to reproduce**:

LogQL with multiple queries

**Actual Result**:

All log lines are now the same color

**Expected Result**

Colored lines

**Environment**:
- Grafana version: 7.5.2

**Comment by hugohaggmark** [comment]
Maybe someone in @grafana/observability-squad knows which issue @Morriz is referring to?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
